delayrecorder.py

```diff
@@ -1,87 +1,175 @@
+            if i == 2:
+                # convert every string in the array into a float
+                stats_list = [float(i) for i in row]
+                # grab each element of the list to create the Stats object
+                stats = Stats(*stats_list)
+    print("Obtained values from " + filename)
+
+    return measurements, stats
+
+def generate_stats(measurements):
+    # Obtain general stats
+    measurements_np = np.array(measurements)
+    min = np.min(measurements_np)
+    max = np.max(measurements_np)
+    mean = np.mean(measurements_np)
+    median = np.median(measurements_np)
+    std = np.std(measurements_np)
+    print("Min/Max G2G Delay: " + '%.2f' % (min) + "/" + '%.2f' % (max) + " ms")
+    print("Mean/Median/Std G2G Delay: " + '%.2f' % (mean) + "/" + '%.2f' % (median) + "/" + '%.2f' % (std) + " ms")
+
+    return measurements_np, Stats(len(measurements_np), min, max, mean, median, std)
+
+def plot_results(measurements, stats, figName):
+    plt.hist(measurements, bins=20)
+    plt.gcf().canvas.manager.set_window_title(figName)
+    plt.title('Latency Histogram')
+    plt.xlabel('Latency (ms)')
+    plt.ylabel('Frequency')
+
+    ax = plt.gca()
+    props = dict(boxstyle='round', facecolor='wheat', alpha=0.5)
+    textstr1 = '\n'.join((
+        r'$\mathrm{min}=%.2f$' % (stats.min, ),
+        r'$\mathrm{max}=%.2f$' % (stats.max, )))
+    # place it at position x=0.05, y=0.95, relative to the top and left of the box
+    ax.text(0.05, 0.95, textstr1, transform=ax.transAxes, fontsize=14,
+        verticalalignment='top', horizontalalignment='left', bbox=props)
+    textstr2 = '\n'.join((
+        r'$\mu=%.2f$' % (stats.mean, ),
+        r'$\mathrm{median}=%.2f$' % (stats.median, ),
+        r'$\sigma=%.2f$' % (stats.std, )))
+    # place it at position x=0.95, y=0.90, relative to the top and right of the box
+    ax.text(0.95, 0.95, textstr2, transform=ax.transAxes, fontsize=14,
+        verticalalignment='top', horizontalalignment='right', bbox=props)
+
+    plt.savefig(figName)
+    print("Saved histogram to " + figName)
+    plt.show()
+
+if __name__ == "__main__":
+    # Set up signal handler to handle keyboard interrupts
+    signal.signal(signal.SIGINT, sigint_handler)
+
+    # Parse input arguments
+    parser = argparse.ArgumentParser(description="This script obtains delay measurements from a Glass-to-Glass device connected via USB, saves results and statistics to a CSV file, and displays the results in a histogram plot. You can also use it to display results from saved measurements in a previous test, in which case the CSV file is read from rather than written to.")
+
+    parser.add_argument('filename', nargs='?', default="results.csv", type=str, help="The name of the CSV file where the data is saved. It is also the name for the saved plot. Default is 'results.csv' which will cause 'results.png' to be created as well. This is the filename used when using the '--readcsv' option as well.")
```

Comment:
`nargs` means that there can be multiple args. Not here, this can be given or not.

```suggestion
    parser.add_argument('filename', default="results.csv", type=str, help="The name of the CSV file where the data is saved. It is also the name for the saved plot. Default is 'results.csv' which will cause 'results.png' to be created as well. This is the filename used when using the '--readcsv' option as well.")
```

Reply:
Similar below